### DISCUSSION TITLE
Have RAG on by default

### DISCUSSION THREAD

**Discussion by mikael1234** [open-webui/open-webui#5848] [General]
Is is possible to setup a set of documents that are used as RAG context by default, without user having to select any docs with #?

**Comment by flefevre** [comment]
Yes, you have to create your own assistant in Model Assistant and to assocaite to it a collection athat you have preloaded.
make it sense for you?

**Reply by mikael1234** [reply]
That was an easy solution, thanks!

**Comment by sir3mat** [comment]
Hi, i have seen a strange behaviour using rag and pipelines

The workflow is as follows:

    Documents are loaded and fed into OpenWebUI.
    OpenWebUI encodes the documents, splits them into chunks, and stores them.
    The /inlet endpoint on the pipeline is called.
    OpenWebUI then performs the RAG operation.
    The pipeline’s /pipe endpoint is triggered, displaying the response on OpenWebUI.
    Finally, the client calls the /outlet endpoint

Question on Implementation:

Why do the /inlet, /pipe, and /outlet endpoints each have different request bodies?
Why the process is openwebuiClient->inlet->openwebuiClient->pipe->openwebuiClient->outlet->openwebuiClient?

If I develop RAG on my custom pipeline in pipe logic tha RAG of openwebui is still executed.
